add 'repo' install method to fetch and install package directly from packagecloud

rplessl opened the pull request:
This add's an additional install method 'repo' which includes the packagecloud repositories for install and update grafana. Enhanced for Debian/Ubuntu and RedHat.

Also fix package name of libfontconfig to libfontconfig1 which is the correct one on Ubuntu LTS. This fixes the reinstallation of libfontconfig each time the puppet runs.

spec file is enhanced, but needs a review from a more experienced spec file writer.

adamcstephens commented:
Many modules add an enable_repo parameter. This allows people who create mirrors or packages to opt out of the module managing the repo.

bfraser commented:
Hi @rplessl thank you for the pull request! I'd like to merge this, though I have some comments / questions.

1. I am in agreement with @adamcstephens that we should provide the ability for the module to **not** manage the package repository, in case the user has their own local repository they are maintaining.

2. Have you tested this code on Red Hat systems? I ask because I don't believe [this code](https://github.com/rplessl/puppet-grafana/blob/master/manifests/install.pp#L75-77) will work as the package name is `fontconfig`.

3. There is no ordering of the [package resources](https://github.com/rplessl/puppet-grafana/blob/master/manifests/install.pp#L75-80) via the `require` metaparameter, so there is the possibility of the module attempting to install Grafana before `libfontconfig1` (or `fontconfig`, as the case may be).

4. Items 2 and 3 are not caught by the tests in the [`repo install method`](https://github.com/rplessl/puppet-grafana/blob/master/spec/classes/grafana_spec.rb#L74-111) context because the `install_method` parameter is not being set to `repo` and it is falling back to the default, `package`. See the [code](https://github.com/rplessl/puppet-grafana/blob/master/spec/classes/grafana_spec.rb#L114-116) in the `archive install method` context for an example of this.

Again, thanks for the submission!

rplessl commented:
Hi @bfraser!

Thanks for your quick reply.

1. I agree, there should be a way for local repositories ... I will change my code

2. Not tested with RedHat System, I agree, the marked install section should be moved to plattform specific part ... will follow in the next commit.

3. Agree, will fix that in the next commit.

4. Ok, will fix that too.

Thanks for the review! A new submission of the pull request will follow.

rplessl commented:
Hi @bfraser @adamcstephens,

I have added the suggested points:

Finding 1 / 4: There exist now a configuration value manage_package_repo to enable / disable the packagecloud repository.

Finding 2 / 3: Fixed.

Thanks for review again!

bfraser commented:
Hi @rplessl this looks good. Thank you, I appreciate the contribution!
